webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/DataSourceService.scala

```diff
@@ -86,6 +88,75 @@ class DataSourceService @Inject()(
     } yield ()
   }
 
+  private def sendRealPaths(dataSources: List[InboxDataSource]) =
+    for {
+      _ <- Fox.successful(())
+      magPathBoxes = dataSources.map(ds => (ds, determineMagRealPathsForDataSource(ds)))
+      pathInfos = magPathBoxes.map {
+        case (ds, Full(magPaths)) => DataSourcePathInfo(ds.id, magPaths)
+        case (ds, Failure(msg, _, _)) =>
+          logger.error(s"Failed to determine real paths for mags of $ds: $msg")
```

Comment:
This should use formatFailureChain from the Formatter trait. Have a look at the other usages.